In [2]:
import numpy as np
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

In [3]:
raw_data = pd.read_csv("2.02.+Binary+predictors.csv")
raw_data.head()

SAT Admitted  Gender
0  1363       No    Male
1  1792      Yes  Female
2  1954      Yes  Female
3  1653       No    Male
4  1593       No    Male

In [4]:
dt = raw_data.copy()
dt["Admitted"] = dt["Admitted"].map({"Yes":1 , "No":0 })
dt["Gender"] = dt["Gender"].map({"Male":0, "Female":1})
dt.head()

SAT  Admitted  Gender
0  1363         0       0
1  1792         1       1
2  1954         1       1
3  1653         0       0
4  1593         0       0

In [5]:
y = dt["Admitted"]
#x1 = dt[["SAT", "Gender"]]
x1 = dt["Gender"]

In [6]:
x = sm.add_constant(x1)
reg = sm.Logit(y, x)
result = reg.fit()
result.summary()

Optimization terminated successfully.
         Current function value: 0.572260
         Iterations 5


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:               Admitted   No. Observations:                  168
Model:                          Logit   Df Residuals:                      166
Method:                           MLE   Df Model:                            1
Date:                Sun, 16 Jul 2023   Pseudo R-squ.:                  0.1659
Time:                        21:12:09   Log-Likelihood:                -96.140
converged:                       True   LL-Null:                       -115.26
Covariance Type:            nonrobust   LLR p-value:                 6.283e-10
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.6436      0.222     -2.901      0.004      -1.078      -0.209
Gender         2.0786      0.363      5.727      0.000       1.367       2.790
==============================================================================
"""

- #### log(odds) = -0.64 + 2.08 * Gender

       log(odds2) = -0.64 + 2.08 * Gender2
       log(odds1) = -0.64 + 2.08 * Gender1
       =
       log(odds2/odds1) = 2.08(Gender2-Gender1) -> Gender has only 1 or 0 options
       =
       log(oddsFemale / oddsMale) = 2.08 * (1 - 0)
       =
       oddsFemale = 7.99 * oddsMale
       


# New regression

In [8]:
y = dt["Admitted"]
x1 = dt[["SAT", "Gender"]]

In [9]:
x = sm.add_constant(x1)
reg = sm.Logit(y, x)
result = reg.fit()
result.summary()

Optimization terminated successfully.
         Current function value: 0.120117
         Iterations 10


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:               Admitted   No. Observations:                  168
Model:                          Logit   Df Residuals:                      165
Method:                           MLE   Df Model:                            2
Date:                Sun, 16 Jul 2023   Pseudo R-squ.:                  0.8249
Time:                        21:17:58   Log-Likelihood:                -20.180
converged:                       True   LL-Null:                       -115.26
Covariance Type:            nonrobust   LLR p-value:                 5.118e-42
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
const        -68.3489     16.454     -4.154      0.000    -100.598     -36.100
SAT            0.0406      0.010      4.129      0.000       0.021       0.060
Gender         1.9449      0.846      2.299      0.022       0.287       3.603
==============================================================================

Possibly complete quasi-separation: A fraction 0.27 of observations can be
perfectly predicted. This might indicate that there is complete
quasi-separation. In this case some parameters will not be identified.
"""

- #### log-likelihood is higher which is better model

- #### for same SAT score, a female has 7(exp(1.9449)) times higher odds to get admitted